server/data/cycles.ts

```diff
@@ -0,0 +1,137 @@
+import { cycles, users } from '../config/mongoCollections';
+import { ObjectId } from 'mongodb';
+import { checkObjectId } from '../helpers/error';
+
+export default {
+
+    /**
+     * @description Get a cycle by id
+     * @param {string} id Cycle id
+     * @returns {Promise<Object>} Cycle object
+     */
+    readById: async (id: string): Promise<object> => {
+        if(!checkObjectId(id)){
+            throw new Error('Invalid id');
+        }
+
+        const cycleCollection = await cycles();
+        const cycle = await cycleCollection.findOne({_id: new ObjectId(id)});
+        if(cycle === null)
+            throw new Error("There is no cycle with that id.");
+
+        cycle['_id'] = cycle['_id'].toString();
```

Comment:
not sure if we will do this, since doing this will change the type of the object